webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -532,32 +525,17 @@ const AdvancedTextEditor = ({
         }
     }, [hasDraftMessage]);
 
-    const handleMouseUpKeyUp = useCallback((e: React.MouseEvent | React.KeyboardEvent) => {
-        setCaretPosition((e.target as TextboxElement).selectionStart || 0);
-    }, []);
-
-    const prefillMessage = useCallback((message: string, shouldFocus?: boolean) => {
+    const prefillMessage = useCallback((message: string) => {
         handleDraftChange({
             ...draft,
             message,
         });
-        setCaretPosition(message.length);
 
-        if (shouldFocus) {
-            const inputBox = textboxRef.current?.getInputBox();
-            inputBox?.click();
-            focusTextbox(true);
-        }
+        const inputBox = textboxRef.current?.getInputBox();
+        inputBox?.click();
+        focusTextbox(true);
     }, [handleDraftChange, focusTextbox, draft, textboxRef]);
 
-    // Update the caret position in the input box when changed by a side effect
-    useEffect(() => {
-        const textbox: HTMLInputElement | HTMLTextAreaElement | undefined = textboxRef.current?.getInputBox();
-        if (textbox && textbox === document.activeElement && textbox.selectionStart !== caretPosition) {
-            Utils.setCaretPosition(textbox, caretPosition);
```

Comment:
This is the main cause of these issues along with `handleMouseUpKeyUp` updating the `caretPosition` on every click or key press